web_editor/js/cards.js

```diff
@@ -1,13 +1,47 @@
-export function makeLaneCard(lane) {
+export function makeLaneCard(lane, idx, app) {
   var node = document.createElement("div");
-  node.setAttribute("class", "card");
-  node.setAttribute("title", JSON.stringify(lane, null, 2));
-  node.setAttribute("style", `background: ${backgroundColor(lane)};`);
-  node.laneJSON = lane;
+  node.className = "card";
+  node.title = JSON.stringify(lane, null, 2);
+  node.style = `background: ${backgroundColor(lane)};`;
 
-  node.innerHTML = `<div align="center">` + typeIcon(lane) + `</div>`;
-  node.innerHTML += `<div align="center">` + directionIcon(lane) + `</div>`;
-  node.innerHTML += `<div align="center">` + width(lane) + `</div>`;
+  node.innerHTML = `<div align="center">${typeIcon(lane)}</div>`;
+  node.innerHTML += `<div align="center">${directionIcon(lane)}</div>`;
```

Comment:
*Assigning* to `innerHTML` is fine, it's just the *append* that is problematic, as described in this answer: https://stackoverflow.com/a/23339072

I think the template literals approach is a good style, just commit it to the DOM in one go :)

Reply:
Ahhhh thank you, I understand now. Serializing to HTML and then parsing everything out again sounds awful!

I'll keep things as all objects for now, because the next step will need to make the direction arrow clickable. AFAICT, there's not a nice way to mix template literals and attaching callbacks, without assigning an ID and then looking it up, or [making one callback / listener at the parent level](https://stackoverflow.com/a/31876859)

Reply:
Using listeners on some parent can work well, especially using [`data-` attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) instead of `id`s to communicate data.